schemes/check_energy/check_energy_chng.F90

```diff
@@ -0,0 +1,414 @@
+    if(is_first_timestep) then
+      teout(:ncol) = te_ini_dyn(:ncol)
+    endif
+
+  end subroutine check_energy_chng_timestep_init
+
+
+  ! Check that energy and water change matches the boundary fluxes.
+!> \section arg_table_check_energy_chng_run Argument Table
+!! \htmlinclude arg_table_check_energy_chng_run.html
+  subroutine check_energy_chng_run( &
+       ncol, pver, pcnst, &
+       q, pdel, &
+       u, v, T, &
+       pintdry, phis, zm, &
+       cp_phys, &              ! cpairv generally, cpair fixed size for subcolumns code
+       cp_or_cv_dycore, &
+       scaling_dycore, &       ! From check_energy_scaling to work around subcolumns code
+       te_cur_phys, te_cur_dyn, &
+       tw_cur, &
+       tend_te_tnd, tend_tw_tnd, &
+       temp_ini, z_ini, &
+       count, ztodt, &
+       latice, latvap, &
+       energy_formula_physics, energy_formula_dycore, &
+       name, flx_vap, flx_cnd, flx_ice, flx_sen, &
+       errmsg, errflg)
+
+    ! Dependency for hydrostatic energy calculation (physics and dycore formulas)
+    use cam_thermo,         only: get_hydrostatic_energy
+
+    ! Dependency for energy formula used by physics and dynamical cores
+    use cam_thermo_formula, only: ENERGY_FORMULA_DYCORE_FV, ENERGY_FORMULA_DYCORE_SE, ENERGY_FORMULA_DYCORE_MPAS
+
+    ! FIXME hplin: for DEBUG only
+    use cam_logfile,        only: iulog
+
+    ! Input arguments
+    integer,            intent(in)    :: ncol           ! number of atmospheric columns
+    integer,            intent(in)    :: pver           ! number of vertical layers
+    integer,            intent(in)    :: pcnst          ! number of ccpp constituents
+    real(kind_phys),    intent(in)    :: q(:,:,:)       ! constituent mass mixing ratios [kg kg-1]
+    real(kind_phys),    intent(in)    :: pdel(:,:)      ! layer thickness [Pa]
+    real(kind_phys),    intent(in)    :: u(:,:)         ! zonal wind [m s-1]
+    real(kind_phys),    intent(in)    :: v(:,:)         ! meridional wind [m s-1]
+    real(kind_phys),    intent(in)    :: T(:,:)         ! temperature [K]
+    real(kind_phys),    intent(in)    :: pintdry(:,:)   ! interface pressure dry [Pa]
+    real(kind_phys),    intent(in)    :: phis(:)        ! surface geopotential [m2 s-2]
+    real(kind_phys),    intent(in)    :: zm(:,:)        ! geopotential height at layer midpoints [m]
+    real(kind_phys),    intent(in)    :: temp_ini(:,:)  ! initial temperature [K]
+    real(kind_phys),    intent(in)    :: z_ini(:,:)     ! initial geopotential height [m]
+    real(kind_phys),    intent(in)    :: cp_phys(:,:)   ! enthalpy (cpairv generally) [J kg-1 K-1]
+    real(kind_phys),    intent(in)    :: cp_or_cv_dycore(:,:)  ! enthalpy or heat capacity, dycore dependent [J K-1 kg-1]
+    real(kind_phys),    intent(in)    :: scaling_dycore(:,:)   ! scaling for conversion of temperature increment [1]
+    real(kind_phys),    intent(in)    :: ztodt          ! 2 delta t (model time increment) [s]
```

Comment:
I believe `ztodt` is just the standard model physics timestep (i.e. I think the `2 delta t` was from back when the Eulerian dycore was used, which had a different time discretization, but is now no longer correct).

Reply:
Thanks, changed comment to just `physics timestep [s]`